packages/interactor/types/locators.ts

```diff
@@ -0,0 +1,43 @@
+import { createInteractor } from '../src/index';
+
+const Link = createInteractor<HTMLLinkElement>('link')({
+  selector: 'a',
+  defaultLocator: ['byHref', 'byTitle'],
+  locators: {
+    byHref: (element) => element.href,
+    byTitle: (element) => element.title
+  }
+});
+
+Link('foobar');
+
+Link.byHref('foobar');
+
+// cannot use wrong type argument on locator
+// $ExpectError
+Link.byHref(123);
+
+// cannot use locator which is not defined
+// $ExpectError
+Link.byValue('foobar');
+
+const Button = createInteractor<HTMLButtonElement>('button')({
+  selector: 'button',
+  defaultLocator: 'byText',
+  locators: {
+    byText: (element) => element.textContent || ""
+  }
+});
+
+Button('foobar');
+
+const TextField = createInteractor<HTMLInputElement>('text field')({
+  selector: 'input[type="text"]',
+  // $ExpectError
+  defaultLocator: 'byLabel',
```

Comment:
@dagda1 This is not erroring as I'd like it to. `byLabel` is not in the map of `locators` below, so I want that to be a type error.